lib/std/fmt.zig

```diff
@@ -1915,6 +1915,54 @@ test "parseUnsigned" {
     try std.testing.expectError(error.InvalidCharacter, parseUnsigned(u8, "", 10));
 }
 
+/// Parses a number like '2G', '2Gi', or '2GiB'.
+pub fn parseIntSizeSuffix(buf: []const u8, radix: u8) ParseIntError!usize {
+    var without_b = buf;
+    if (mem.endsWith(u8, buf, "B")) without_b.len -= 1;
+    var without_i = without_b;
+    var base: usize = 1000;
+    if (mem.endsWith(u8, without_b, "i")) {
+        without_i.len -= 1;
+        base = 1024;
+    }
+    if (without_i.len == 0) return error.InvalidCharacter;
+    const orders_of_magnitude: usize = switch (without_i[without_i.len - 1]) {
+        'k', 'K' => 1,
```

Comment:
Multiple ways of doing this doesn't feel necessary, but not sure if hackability is more important in this case.

Reply:
You mean the k/K? I did that because the SI and IEC use different capitalizations for the K, which I learned from reading the fmtIntSize docstrings.